openshift/buildConfig-template.yaml

```diff
@@ -0,0 +1,81 @@
+apiVersion: v1
+kind: Template
+metadata:
+  name: package-analyzer-buildconfig
+  annotations:
+    description: This is Thoth Core - Package Analyzer BuildConfig
+    openshift.io/display-name: "Thoth: Package Analyzer BuildConfig"
+    version: 0.3.0
+    tags: poc,thoth,ai-stacks,package-analyzer
+    template.openshift.io/documentation-url: https://github.com/Thoth-Station/
+    template.openshift.io/long-description: |-
+      This template defines resources needed to deploy Thoth Package
+      Analyzer as a Proof-of-Concept to OpenShift.
+    template.openshift.io/provider-display-name: Red Hat, Inc.
+  labels:
+    template: package-analyzer-buildconfig
+    app: thoth
+    component: package-analyzer
+
+objects:
+  - kind: BuildConfig
+    apiVersion: v1
+    metadata:
+      name: package-analyzer
+      labels:
+        app: thoth
+        component: package-analyzer
+    spec:
+      resources:
+        requests:
+          cpu: 2
+          memory: 768Mi
+        limits:
+          cpu: 2
+          memory: 1024Mi
+      output:
+        to:
+          kind: ImageStreamTag
+          name: "package-analyzer:${IMAGE_STREAM_TAG}"
+      source:
+        type: Git
+        git:
+          uri: "${GITHUB_URL}"
+          ref: "${GITHUB_REF}"
+      strategy:
+        type: Source
+        sourceStrategy:
+          from:
+            kind: ImageStreamTag
+            name: python-36-centos7:latest
+          env:
+            - name: APP_FILE
+              value: "thoth-package-analyzer"
```

Comment:
Is this configuration relevant? This is build config.

Reply:
@fridex isn't it looking for the file name passed to Python interpreter which is responsible for launching the app which is in our case **thoth-package-analyzer**? I took a look at thot-adviser's buildConfig [1] and there is a doc [2].

[1] https://github.com/thoth-station/adviser/blob/master/openshift/buildConfig-template.yaml#L78
[2] https://docs.openshift.com/container-platform/3.6/using_images/s2i_images/python.html

Reply:
These configuration entries are for the actual deployment/job template. It has nothing to do with builds. Opened a PR to fix this in the adviser repo, PTAL. https://github.com/thoth-station/adviser/pull/321

Reply:
Thanks @fridex , it is good to know!